share-on-mastodon.js

```diff
@@ -197,120 +208,143 @@ export class shareOnMastodon extends HTMLElement {
+        this.getAttribute("data-default-url");
+    }
 
     // Get the share button text from attribute if available
-    if (this.getAttribute('share_text')) {
-      this.shadowRoot.querySelector('#shareText').innerHTML = this.getAttribute('share_text');
-    };
+    if (this.getAttribute("data-share-text")) {
+      this.shadowRoot.querySelector("#share-text").innerHTML =
+        this.getAttribute("data-share-text");
+    }
   }
 
   openModal() {
-  // Prefill the form with the user's previously-specified Mastodon instance, if applicable
-  if (localStorage['mastodon_instance']) {
-    this.shadowRoot.querySelector('#instanceInput').value = localStorage['mastodon_instance'];
-  }
+    // Prefill the form with the user's previously-specified Mastodon instance, if applicable
+    const storedInstance = localStorage?.getItem("mastodonInstance");
+    if (storedInstance) {
+      this.shadowRoot.querySelector("#instance-input").value = storedInstance;
+    }
 
-  // open Dialog as modal
-  this.shadowRoot.querySelector('#mastodonModal').showModal();
+    // open Dialog as modal
+    this.shadowRoot.querySelector("#mastodon-modal").showModal();
   }
 
-  validate_before_share() {
-    if (this.shadowRoot.querySelector('#instanceInput').checkValidity()) {
-      this.share_on_mastodon();
+  urlCheck() {
+    let val = this.shadowRoot.querySelector("#instance-input").value;
+    if (!val == "" && !val.startsWith("https://")) {
+      this.shadowRoot.querySelector("#instance-input").value = "https://" + val;
     }
+    this.shadowRoot.querySelector("#instance-input").checkValidity();
   }
 
-  share_on_mastodon() {
+  shareOnMastodon() {
     // Set instance from the modal text input
-    let instance = this.shadowRoot.querySelector('#instanceInput').value;
+    let instance = this.shadowRoot.querySelector("#instance-input").value;
 
     // check, format, and cache instance
     if (instance) {
       // Handle URL formats
-      if ( !instance.startsWith("https://") && !instance.startsWith("http://") )
+      if (!instance.startsWith("https://") && !instance.startsWith("http://"))
         instance = "https://" + instance;
 
       // Handle ending slash
-      if ( !instance.endsWith("/") )
-        instance = instance + "/";
+      if (!instance.endsWith("/")) instance = instance + "/";
 
       // Cache the instance/domain for future requests
-      localStorage['mastodon_instance'] = instance;
-    };
+      localStorage?.setItem("mastodonInstance", instance);
+    }
 
-    // If the share_title attribute is set use it. Otherwise, use the page title
+    // If the share-title attribute is set use it. Otherwise, use the page title
     let shareTitle = this.pageTitle;
-    if (this.getAttribute('share_title')) {
-      shareTitle = this.getAttribute('share_title')  + "\n";
+    if (this.getAttribute("data-share-title")) {
+      shareTitle = this.getAttribute("data-share-title") + "\n";
```

Comment:
Instead of using `this.getAttribute("something")` you can make use of the [dataset property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) lie `this.dataset.something`

Reply:
Ah, right you are. I will update this. Thanks!

Reply:
updated ✅